cloudinit/log.py

```diff
@@ -22,6 +22,7 @@
 
 # Always format logging timestamps as UTC time
 logging.Formatter.converter = time.gmtime
+logging.raiseExceptions = True
```

Comment:
This is a pretty big change in behavior, isn't it? The patcher code made it such that if an exception was raised and the handler couldn't handle it (i.e., failure to write to log file), then we'd instead send the log messages to stderr.

It looks like this is just raising the file write exception. We likely lose the original log and cloud-init dies because it couldn't write to the log file.

Did I read that correctly?

Reply:
> This is a pretty big change in behavior, isn't it? The patcher code made it such that if an exception was raised and the handler couldn't handle it (i.e., failure to write to log file), then we'd instead send the log messages to stderr.
> 
> It looks like this is just raising the file write exception. We likely lose the original log and cloud-init dies because it couldn't write to the log file.

Not quite how that works, the exception would just be logged, but I agree that it's not quite the behavior we want since the stderr output could be helpful for debugging, if for example running cloud-init interactively. I'll update this with something better.